src/main/java/com/kodilla/ecommercee/entity/Cart.java

```diff
@@ -16,15 +16,15 @@
 public class Cart {
 
     @Id
-    @GeneratedValue
+    @GeneratedValue(strategy = GenerationType.AUTO)
     @EqualsAndHashCode.Include
     private Long id;
 
     @NonNull
-    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+    @OneToMany(cascade = CascadeType.MERGE)
```

Comment:
Tu brakuje (mappedBy = "cart"). MERGE nie działa i próbuje ustawić w rekordach item pole cart_id=null, a ponieważ my zdefiniowaliśmy to pole na NotNull, więc powstaje konflikt i błąd. Tu trzeba dać REMOVE i usunąć jeden błąd w teście public void testCartAndItem() -> na samym końcu wykasować 'cartRepository.deleteById(cart.getId())'
LUB:
usunąć cascadowość w item
w ItemRepository dodać 'void deleteItemsByCart(Cart cart);' i w serwisie przed deleteCartById() wywołać ją (usunie najpierw po kolei wszystkie itemy)
LUB
usunąć cascadowość
w ItemRepository dodać namedQuery:
    @Modifying
    @Query(value = "Delete from Item WHERE cart_id = :cartId")
    void deleteItemsByCartId(@Param("cartId") Long id);
i potem tak samo przed cartRepository.deleteById() ja wywołać (usunie od razu wszystkie itemy)